src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/actionfilter/FieldCapsFilter.kt

```diff
@@ -33,57 +34,77 @@ import org.elasticsearch.action.support.ActionFilterChain
+            flag -> shouldIntercept = flag
+        }
+    }
+
     override fun <Request : ActionRequest?, Response : ActionResponse?> apply(
         task: Task,
         action: String,
         request: Request,
         listener: ActionListener<Response>,
         chain: ActionFilterChain<Request, Response>
     ) {
-        if (request is FieldCapabilitiesRequest) {
+        if (request is FieldCapabilitiesRequest && shouldIntercept) {
             val indices = request.indices().map { it.toString() }.toTypedArray()
-            val concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterService.state(), request.indicesOptions(), *indices)
             val rollupIndices = mutableSetOf<String>()
             val nonRollupIndices = mutableSetOf<String>()
-            for (index in concreteIndices) {
-                val isRollupIndex = RollupSettings.ROLLUP_INDEX.get(clusterService.state().metadata.index(index).settings)
-                if (isRollupIndex) {
-                    rollupIndices.add(index)
-                } else {
-                    nonRollupIndices.add(index)
+            val remoteClusterIndices = GuiceHolder.remoteClusterService.groupIndices(request.indicesOptions(), indices) {
+                idx: String? -> indexNameExpressionResolver.hasIndexAbstraction(idx, clusterService.state())
+            }
+            val localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)
+
+            localIndices?.let {
+                val concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterService.state(), request.indicesOptions(), it)
+                for (index in concreteIndices) {
+                    val isRollupIndex = RollupSettings.ROLLUP_INDEX.get(clusterService.state().metadata.index(index).settings)
+                    if (isRollupIndex) {
+                        rollupIndices.add(index)
+                    } else {
+                        nonRollupIndices.add(index)
+                    }
+                }
+            }
+
+            remoteClusterIndices.entries.forEach {
+                val cluster = it.key
+                val clusterIndices = it.value
+                clusterIndices.indices().forEach { index ->
+                    nonRollupIndices.add("$cluster${RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR}$index")
                 }
             }
+            logger.debug("Resolved into rollup $rollupIndices and non rollup $nonRollupIndices indices")
 
             if (rollupIndices.isEmpty()) {
                 return chain.proceed(task, action, request, listener)
             }
 
-            if (nonRollupIndices.isEmpty()) {
-                val rewrittenResponse = rewriteResponse(mapOf(), arrayOf(), rollupIndices)
-                return listener.onResponse(rewrittenResponse as Response)
+            if (nonRollupIndices.isNotEmpty()) {
```

Comment:
I will add some documentation here, I think this is not obvious and easy to introduce a bug